libs/dyn/merge/override.go

```diff
@@ -111,11 +117,16 @@ func overrideMapping(basePath dyn.Path, leftMapping dyn.Mapping, rightMapping dy
 		if _, ok := rightMapping.GetPair(leftPair.Key); !ok {
 			path := basePath.Append(dyn.Key(leftPair.Key.MustString()))
 
-			err := visitor.VisitDelete(path, leftPair.Value)
+			deleteOut, err := visitor.VisitDelete(path, leftPair.Value)
 
 			if err != nil {
 				return dyn.NewMapping(), err
 			}
+
+			// if 'delete' was undone, add it back
+			if deleteOut != dyn.NilValue {
```

Comment:
You need to check the `Kind()`. A nil value may have a location attached to it so direct equality can fail.